### ISSUE TITLE
Fonts.com version is hanging with the class wf-loading, no events called

### ISSUE THREAD

**Issue by petebarr** [typekit/webfontloader#331]
I had this working on a live site for a while but recently it stopped calling the active function.

I have a pen here to simulate the live site (which I'm going to remove webfontloader in the meantime). If you look in the console you will see only 'loading' being called. https://codepen.io/petebarr/pen/RRrNqm

**Comment by pburke** [comment]
I'm seeing this too.

**Comment by pburke** [comment]
I was able to fix this issue by updating to the latest version of webfontloader.

**Comment by lexbi** [comment]
@pburke but he is using the most recent version...

I am also experiencing this problem. I am using: 
`wf.src = 'https://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js';`

Which gets 1.6.16 There is no higher version at this time hosted by googleapis.com.

**Comment by lexbi** [comment]
It appears that the problem is with fonts.com. If I comment out my projectId within the `monotype` object, the `.wf-loading` class gets removed from the `html` and `.wf-active` class is added.

**Comment by dkozma** [comment]
I'm running 1.6.26 via NPM and running into this issue as well. Per @lexbi's comment, I've reached out to Monotype as well to make sure their support staff is aware of this problem.

**Comment by petebarr** [comment]
I got the follow response from Monotype:

_Hello Peter,

I apologize for your inconvenience. It sounds like you might be using an older version of the webfont loader API (if you are using the Google hosted version, it is very outdated at 1.6.16). You can use an up-to-date version of the API by hosting the JavaScript file with your website itself or by using the CloudFlare CDN's hosted version:

https://github.com/typekit/webfontloader 

https://cdnjs.com/libraries/webfont 

Additionally, if you are using a node-based backend, it may interest you to know that the maintainer of the GitHub repo also maintains an npm package: 

https://www.npmjs.com/package/webfontloader 

Let us know if you have any further questions.

Best Regards,

Domenic_

**Comment by dkozma** [comment]
@petebarr: Looking at the `gitHead` SHA in my `/node_modules/webfontloader/package.json`, I'm seeing 33bec9da7ca4b0a46b1efbaca705c2ae5029840b which corresponds to the release of v1.6.26, and still running into this issue (the `loading` callback works, but `active` does not fire).

I've e-mailed Domenic that this was the case as well.

**Comment by petebarr** [comment]
@dkozma Yeah, I'm getting a timeout this time round. Not sure if this might be down to the iframe in codepen for fonts.com, but the typekit version works perfectly.

**Comment by dkozma** [comment]
For me, the fonts load, but the `active` event doesn't fire. The library is also correctly getting a 200 when loading the font resource.

I tried to debug the webfontloader library, however it seems that NPM is pointing to the minified `/webfontloader.js` file which is hard to put debug statements into. Unfortunately, there's not much documentation on how to easily modify/debug the source files (that I can find, at least).

**Comment by lexbi** [comment]
Just requested google update their hosted libraries here: https://groups.google.com/forum/#!topic/google-ajax-search-api/SpsP5bmc_pc

No idea if any google staff actually view that forum, but, hopefully someone will suggest where else we could submit that request if not.

**Comment by dkozma** [comment]
Monotype's customer support got back to me with the following response (which is pretty awesome of them since this library probably isn't an officially supported means of integration):

> Yes, this is something that our developers are aware of. If you'll notice in the CodePen, there is a "loadAllFonts" property that is commented out. If you enable that property, you will find the commonly expected behavior in regards of which events fire.

> The gist of the issue is that our service's JavaScript is fairly specialized in that it will only include web fonts from the project if it detects that the page is actually using them. For example, if a project has three web fonts included but the JavaScript only detects that the page is trying to use one of them, it will only include that one into the page and therefore one will be active and the other two will be inactive.

> This "partial" state (1/1 attempted fonts loaded but also 1/3 total fonts loaded) is resulting in the ambiguous decision of whether to produce the general "active" or "inactive" callbacks (the individual font callbacks function more predictably).

> As such, there are a few ways to proceed if you are set on using the web font loader to include your Fonts.com web font project and provide callbacks:

> - If you are set on using the general/overall events that are provided, then you can ensure that the page is using all web fonts from the project by removing some from the project that are unused, or, more practically, by enabling the "loadAllFonts" property in the WebFontLoader config block

> - If you can, it may be worthwhile to ignore the general/overall events provided and instead monitor for font-specific events. You can then consider keeping track of fonts loaded vs. inactive and react accordingly (e.g. with logic like "If X number of fonts loaded are inactive (failed to load), do Y").

Following their advice, I tried adding `loadAllFonts: true` to my `monotype` config key and after that both the `active` and `fontactive` events fired successfully. Since I only have the fonts I'm using within in my page contained in my project, I don't think this will have an effect on payload size for me, however you might want to keep this in mind if your project is set up differently.

**Comment by petebarr** [comment]
I got the same response, they were very helpful. I'll be trying it tomorrow but great it's working for you.

**Comment by lexbi** [comment]
1.6.26 is now available from google, BUT, doesn't work with the "1" version which I think is deprecated? But still used on google font's snippets.

https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js
https://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js - still 1.6.16

I've asked google to update that too.

**Comment by kuettel** [comment]
On Sun, Aug 7, 2016 at 1:22 PM, joe <notifications@github.com> wrote:

> 1.6.26 is now available from google, BUT, doesn't work with the "1"
> version which I think is deprecated? But still used on google font's
> snippets.
>
> https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js
> https://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js - still 1.6.16
>
> I've asked google to update that too.
>
FWIW, I am seeing v1.6.26 served for https://ajax.googleapis.com/
ajax/libs/webfont/1/webfont.js

$ wget -O - https://ajax.googleapis.com/ajax/libs/webfont/*1*/webfont.js
2>/dev/null | grep 'Web Font Loader' | cut -c -78
/* Web Font Loader v1.6.26 - (c) Adobe Systems, Google. License: Apache 2.0
*/

$ wget -O - https://ajax.googleapis.com/ajax/libs/webfont/*1.6*/webfont.js
2>/dev/null | grep 'Web Font Loader' | cut -c -78
/* Web Font Loader v1.6.26 - (c) Adobe Systems, Google. License: Apache 2.0
*/

$ wget -O - https://ajax.googleapis.com/ajax/libs/webfont/*1.6.26*/webfont.js
2>/dev/null | grep 'Web Font Loader' | cut -c -78
/* Web Font Loader v1.6.26 - (c) Adobe Systems, Google. License: Apache 2.0
*/

**Comment by lexbi** [comment]
Maybe cdn is cached or something, if I get it with curl or wget I still get 1.6.16

**Comment by kuettel** [comment]
On Mon, Aug 8, 2016 at 10:19 AM, joe <notifications@github.com> wrote:

> Maybe cdn is cached or something, if I get it with curl or wget I still
> get 1.6.16
>
Thank you Joe.  I was able to get to the bottom of this.  You are correct
in that the older version is being cached.  When I checked internally, the
request bypassed the cache, thus the different response.

Long story short though, support for aliases e.g. /ajax/libs/webfont/1,
/ajax/libs/webfont/1.6 has been deprecated for quite a while now.  To
pickup the latest version, users will need to reference it directly. e.g.
/ajax/libs/webfont/1.6.26.

While aliases worked out quite well for webfont.js, it was problematic for
other libraries, where changes would end up breaking integrations.  Thus
the larger push to deprecate aliases.

**Comment by lexbi** [comment]
the 1 version now loads 1.6.26 fine, but, I'd still use a link to the specific version
https://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js
